ast/parser_test.go

```diff
@@ -124,29 +124,37 @@ func TestScalarTerms(t *testing.T) {
 	assertParseOneTerm(t, "exponent", "6e7", FloatNumberTerm(6e7))
 	assertParseOneTerm(t, "string", "\"a string\"", StringTerm("a string"))
 	assertParseOneTerm(t, "string", "\"a string u6abc7def8abc0def with unicode\"", StringTerm("a string u6abc7def8abc0def with unicode"))
-	assertParseError(t, "hex", "6abc")
-	assertParseError(t, "non-terminated", "\"foo")
-	assertParseError(t, "non-terminated-raw", "`foo")
-	assertParseError(t, "non-string", "'a string'")
-	assertParseError(t, "non-number", "6zxy")
-	assertParseError(t, "non-number2", "6d7")
-	assertParseError(t, "non-number3", "6\"foo\"")
-	assertParseError(t, "non-number4", "6true")
-	assertParseError(t, "non-number5", "6false")
-	assertParseError(t, "non-number6", "6[null, null]")
-	assertParseError(t, "non-number7", "6{\"foo\": \"bar\"}")
-	assertParseError(t, "non-number8", ".0.")
-	assertParseError(t, "non-number9", "{0e}")
-	assertParseError(t, "non-number9", "{0e.}")
-	assertParseError(t, "non-number10", "{0F}")
-	assertParseError(t, "non-number11", "{00}")
-	assertParseError(t, "non-number12", "{00.1}")
-	assertParseError(t, "non-number13", "{-00}")
-	assertParseError(t, "non-number14", "{-00.1}")
-	assertParseError(t, "non-number14", "{-00.01}")
-	assertParseError(t, "non-number15", "{00e1}")
-	assertParseError(t, "non-number16", "{-00e1}")
-	assertParseError(t, "number too big", "{7e3000000000}")
```

Comment:
Contrary to what one might have thought, that error was triggered by `new(big.Float).SetString(...)` failing, not the exp check. I've adjusted the test cases to differentiate errors.